### ISSUE TITLE
cant get process.env.MODE

### ISSUE THREAD

**Issue by vandelpavel** [freddy38510/quasar-app-extension-ssg#33]
@freddy38510 
I'm trying to split my project and i need it to run half in SSG and the other in SSR.
With that in mind my problem comes when i try to split routes and when i'm asking for the MODE i can't search for ssg.
Also the portion of code that should run in ssr both runs in ssr and ssg.

Any clues on how to separate things?

(also i'm starting to get familiar with routing and dynamic routes) #22

**Comment by freddy38510** [comment]
You're right, you can't combine both modes because i developped this extension with only SSG in mind.

But, the extension should build your app in SSR mode and generate static pages for the routes you provide in SSG options.
It could be interesting to add this feature based on [router option object](https://github.com/freddy38510/quasar-app-extension-ssg/issues/22).

Right now, i think it will be very difficult to handle both modes by yourself. I mean that the app extension will prevent you to do that easily. So, i have to implement this feature. I can't give you a date of release. I will come back to you to let you know how the things progress.

First, i will try to make available `ssg` in `process.env.MODE` quickly.

**Comment by freddy38510** [comment]
@vandelpavel

From [release 1.1.0](https://github.com/freddy38510/quasar-app-extension-ssg/releases/tag/v1.1.0) you can use [`process.env.STATIC`](https://github.com/freddy38510/quasar-app-extension-ssg#processenvstatic)to conditionnaly do things depending if you're app was built with `quasar ssg generate` or not.

**Comment by franco-onevillas** [comment]
Hi @freddy38510, how does the `process.env.STATIC` help with that?

I mean, if you exclude some routes (to work with SSG), is it going to use the SSR on the rest of the routes?

Thank you :)

**Comment by freddy38510** [comment]
Ah, that makes sense, thanks for the explanation.

For now, implementing this feature is not a priority. First, I planned to support Vite.js and lazy hydration via a [new package](https://github.com/freddy38510/vue3-lazy-hydration) I'm working on.

But I will keep your need in mind and let you know when I start making progress.

**Comment by franco-onevillas** [comment]
Thank you Freddy. Wow, that would help with improving the time to Interactive, right?

Cheers

**Comment by freddy38510** [comment]
Yep that's right, in theory it should.

You can also lighten your bundle with code splitting and only load the splitted chunks when hydration occurs.

In practice, splitted chunks are async components inside the `LazyHydrationWrapper` or component wrapped with `import-wrappers` like this:
```js
hydrateNever(() => import('CompA.vue'));
```

By using the library this way with the ssg extension, the js code for the `CompA` will never be loaded at client side.

**Comment by francoromanol** [comment]
That's a good one. I would love to help with these things, but I never dare to get into OSS...
